### PR TITLE
New filter `traefik-auth`

### LINKED ISSUE #2240: Traefik in Docker support
Recently migrated to Docker and Traefik (from plain server and nginx). I can't really locate any reference if fail2ban supports Traefik, except some snippets here: [Traefik fail2ban](https://gist.github.com/acundari/9bdcf2ba0c0f8a4bf59a21d06da35612)
I have the Traefik access logs on my host. I also read that botsearch is not recommended.
Anyway, will fail2ban properly capture events from the logs, and implement the proper firewall rules even if using Docker?
Thanks

### PR DESCRIPTION
**Pull request by crazy-max**
Closes #2240 

Used to ban hosts, that were failed through traefik

### CODE REVIEW

**Comment by sebres** [comment]
Looks like it is a web-server (service) and you want to catch 401 (Unauthorized) failures.
The question is what happens by regular login with legitimate user/password/etc?
Do you see once a single line with 401 in success case also? 
Could you please provide an excerpt with all messages for one successful and one failed login attempt?

**Comment by crazy-max** [comment]
@sebres

> Looks like it is a web-server (service) and you want to catch 401 (Unauthorized) failures.
> The question is what happens by regular login with legitimate user/password/etc?
> Do you see once a single line with 401 in success case also?

Yes indeed, in case of success a single line with 401 appears :

```
10.0.0.2 - - [27/Nov/2018:23:33:30 +0000] "GET /dashboard/ HTTP/2.0" 401 17 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 117 "Auth for frontend-Host-traefik-0" "/dashboard/" 0ms
10.0.0.2 - username [27/Nov/2018:23:33:31 +0000] "GET /dashboard/ HTTP/2.0" 200 716 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 118 "Host-traefik-0" "http://172.16.0.3:8080" 4ms
```

> Could you please provide an excerpt with all messages for one successful and one failed login attempt?

Done

File: fail2ban/tests/files/logs/traefik-auth [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+# failJSON: { "time": "2018-11-18T22:34:34", "match": true , "host": "10.0.0.2" }
+10.0.0.2 - - [18/Nov/2018:21:34:34 +0000] "GET /dashboard/ HTTP/2.0" 401 17 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 72 "Auth for frontend-Host-traefik-0" "/dashboard/" 0ms
```
[the remark this answers is not in the document]
Comment (sebres) [reply fail2ban/tests/files/logs/traefik-auth]: Anyway, if you could provide two log-excerpts (2-3 lines for both success/failed) as a comment, I can make a review and fulfill this PR for you.

File: config/filter.d/traefik-auth.conf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#       - "traefik.frontend.auth.basic.users=test:$$apr1$$H6uskkkW$$IgXLP6ewTrSuBkTrqE8wj/"
+#     volumes:
+#       - "/var/log/traefik:/var/log"
+#       - "/var/run/docker.sock:/var/run/docker.sock"
+#     restart: always
+#
+
+[Definition]
+
+failregex = ^<HOST> \- \S+ \[\] \"(GET|POST|HEAD) .+\" 401 .+$
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply config/filter.d/traefik-auth.conf]: > is there some other message fail2ban can unambiguously identify it as the auth-failure?);

I don't think so. I use this reverse proxy with a lot of services and I have not detected false positives.

> or this regex should be rewritten (something not - in username case);

Indeed, I will push this change

File: fail2ban/tests/files/logs/traefik-auth (hunk 1 above)
[the remark this answers is not in the document]
Comment (crazy-max) [reply fail2ban/tests/files/logs/traefik-auth]:
```
traefik    | 10.0.0.2 - - [14/Dec/2018:17:55:25 +0000] "GET /dashboard/ HTTP/2.0" 401 17 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 29 "Auth for frontend-Host-traefik-0" "/dashboard/" 0ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/ HTTP/2.0" 200 716 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 131 "Host-traefik-0" "http://172.19.0.4:8080" 11ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/styles.e21e21d47be645f690e6.bundle.css HTTP/2.0" 200 116978 "https://traefik.example.com/dashboard/" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 138 "Host-traefik-0" "http://172.19.0.4:8080" 97ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/inline.318b50c57b4eba3d437b.bundle.js HTTP/2.0" 200 796 "https://traefik.example.com/dashboard/" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 143 "Host-traefik-0" "http://172.19.0.4:8080" 70ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/OpenSans-Regular.629a55a7e793da068dc5.ttf HTTP/2.0" 200 217360 "https://traefik.example.com/dashboard/styles.e21e21d47be645f690e6.bundle.css" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 152 "Host-traefik-0" "http://172.19.0.4:8080" 156ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/polyfills.1457c99db4b6dba06e8d.bundle.js HTTP/2.0" 200 59401 "https://traefik.example.com/dashboard/" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 144 "Host-traefik-0" "http://172.19.0.4:8080" 277ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/main.f341693bf9ed22060b42.bundle.js HTTP/2.0" 200 686413 "https://traefik.example.com/dashboard/" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 146 "Host-traefik-0" "http://172.19.0.4:8080" 354ms
traefik    | 10.0.0.2 - username [14/Dec/2018:17:55:26 +0000] "GET /dashboard/scripts.ef668c5c0d42ec4c5e83.bundle.js HTTP/2.0" 200 368525 "https://traefik.example.com/dashboard/" "Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0" 145 "Host-traefik-0" "http://172.19.0.4:8080" 387ms
```

**Comment by crazy-max** [comment]
@sebres Missing merge ?

**Comment by sebres** [comment]
Done, thanks again.

**Pull request merged**